test/resource-limits/mzcompose.py

```diff
@@ -0,0 +1,31 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from materialize.mzcompose import Composition
+from materialize.mzcompose.services import (
+    Kafka,
+    Materialized,
+    Postgres,
+    SchemaRegistry,
+    Testdrive,
+    Zookeeper,
+)
+
+
+SERVICES = [
+    Materialized(),
+    Testdrive(),
+    Postgres(),
+]
+
+
+def workflow_default(c: Composition) -> None:
```

Comment:
This doesn't do anything special. Can `resource_limits.td` just go into test/testdrive and then there's no new needed pipeline?

Reply:
If I understand testdrive correctly then unfortunately I think this needs it's own pipeline. Testdrive doesn't clean up the resources after a test so there's no way for `resource_limits.td` to know what resources already exist.

Reply:
I think you can add `with c.override(Testdrive()):` which will run testdrive and reset its state. It'd have to be its own workflow, so maybe doesn't fit into test/testdrive, but might fit somewhere else? (I'm only poking at this because Nikhil has said how expensive new pipelines are, otherwise I wouldn't care.)

Reply:
Ah that's a good idea. I hadn't thought about putting it into an existing non-testdrive pipeline. I'll find somewhere it fits in.

Reply:
@philip-stoev it looks like we already actually have a similar test framework in `test/limits`. Do you have any thoughts on a good way to combine these two?

The TLDR of this PR is that we are limiting the amount of each resource we create by some pre-configured numbers.

Reply:
The default workflow in test/limits probes for how high the limits are , and expects success, and it is written in a way that does not have a natural place for your .td file. Maybe test/cluster would be a more natural place, as that directory already contains desperate tests that are all called from `workflow_default` ?

With respect to testdrive, it calls `DROP DATABASE` at the start, so all resources except for clusters will be cleaned up at start.

Reply:
Ah that makes sense because it was the cluster limit that was unexpectedly failing.

Reply:
I just moved this to `test/cluster`.